### ISSUE TITLE
Different method signature in Zuul 2 sample and zuul-core 2.1.2

### ISSUE THREAD

**Issue by maregunjob** [Netflix/zuul#594]
Hi,

In the [sample app](https://github.com/Netflix/zuul/tree/2.1/zuul-sample) the [SampleServerStartup](https://github.com/Netflix/zuul/blob/2.1/zuul-sample/src/main/java/com/netflix/zuul/sample/SampleServerStartup.java) file uses the `choosePortsAndChannels` method that it inherits from [BaseServerStartup](https://github.com/Netflix/zuul/blob/2.1/zuul-core/src/main/java/com/netflix/zuul/netty/server/BaseServerStartup.java) and the method takes only 1 parameter, 
i.e. 
`protected abstract Map<Integer, ChannelInitializer> choosePortsAndChannels(ChannelGroup clientChannels);`

However, after downloading the version recommended in the wiki, i.e. 

```
<dependency>
    <groupId>com.netflix.zuul</groupId>
    <artifactId>zuul-core</artifactId>
    <version>2.1.2</version>
</dependency>
```
the method signature in the BaseServerStartup class is different and takes 2 parameters, 
i.e.
`protected abstract Map<Integer, ChannelInitializer> choosePortsAndChannels(ChannelGroup var1, ChannelConfig var2);`

Is the maven artifact built from a different branch?

Thank you.

**Comment by artgon** [comment]
Can you try 2.1.4?

**Comment by maregunjob** [comment]
Thank you for updating the wiki with v2.1.4.

The 2.1.4 sample app on [bintray](https://bintray.com/netflixoss/maven/zuul/2.1.4#files/com%2Fnetflix%2Fzuul%2Fzuul-sample%2F2.1.4) does use the same 2 parameter version of the `choosePortsAndChannels` method as zuul-core 2.1.4 on [bintray](https://bintray.com/netflixoss/maven/zuul/2.1.4#files/com%2Fnetflix%2Fzuul%2Fzuul-core%2F2.1.4). 

However, there's still a mismatch between the 2.1.4 version and the sample app as the sample app the [wiki](https://github.com/Netflix/zuul/wiki/Getting-Started-2.0) points to uses the 1 parameter variant and so does the zuul-core github repository itself. 

I suppose the difference is due to the fact that 2.1.4 was updated 4 months ago, while the github repository has some more recent changes. In fact, this [commit](https://github.com/Netflix/zuul/commit/f582d246af4062f0ea5d1187a5d930de4c28ed96#diff-8b0fb7b302b562f1b2317b21ffd2efd3) changed the method's signature about 2 days ago.

**Comment by artgon** [comment]
Yep, just released 2.1.5 this morning.

**Comment by maregunjob** [comment]
Great. Thank you.
